### DISCUSSION TITLE
Modeling single table strategy

### DISCUSSION THREAD

**Discussion by marceloverdijk** [prisma/prisma#5141] [Q&A] [answered]
I've an existing database schema with a table `Publication` that represents multiple models: `Article` and `BlogPost`. The table contains a discriminator column `type` which indicates whether the publication is a article or a blog post. `Article` and `BlogPost` share common properties (hence a single table strategy is used) but also specific properties per type.

`Publication` also has a foreign key to a `Author` table.
In Prisma I could model this with a relation between `Author` and `Publication` having a `Publication[]` field in the `Author`  model.
But I would prefer to have `Article[]` and `BlogPost[]` fields instead.

Is it possible to model this with Prisma using a single table strategy?

If not only possible solution I see would be creating a `Article`  and `BlogPost`  database view and "connect" these to the `Author` model.

**Comment by ryands17** [comment, marked as answer]
Hey @marceloverdijk 👋 

Prisma's data modelling directly maps to the database so you wouldn't be able to have `Article[]` and `BlogPost[]` as separate entities as they are in the same table and Prisma wouldn't be able to differentiate between the two.

In this case, you can have two possible options:

1. Separate them into two entities at the database level
2. Differentiate them at the application level after you have fetched the data, which I suppose you're doing already.

Let me know if that helps :)

**Reply by marceloverdijk** [reply]
Yes, thx for confirming @ryands17 , this is what I thought.
As database tables are fixed 1. is not an option so indeed differentiating at the application level is what I'm doing now.

I might experiment with creating some database views per type, and adding these manually to the Prisma schema. 
That might work I think. Especially as I'm not concerned to writes; only queries.

**Reply by ryands17** [reply]
Yes that would be a great option as well, but Prisma doesn't natively support views yet, so you would need to use [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) for now as a workaround. Do follow this issue for more updates :)

https://github.com/prisma/prisma/issues/3383

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
